### DISCUSSION TITLE
How do I set the request timeout for the GraphServiceClient?

### DISCUSSION THREAD

**Discussion by andre-qumulo** [microsoftgraph/msgraph-sdk-python#742] [Q&A]
I found this: https://stackoverflow.com/questions/43260659/set-timeout-to-a-higher-value-in-microsoft-graph-sdk

but the http provider doesn't appear to be present in the object.  I can't seem to find the docs on how to set this either.

**Comment by lockness-Ko** [comment]
bump

**Comment by rgajason** [comment]
See: https://github.com/microsoftgraph/msgraph-sdk-python/blob/main/docs/general_samples.md#2-creating-a-graph-client-using-a-custom-httpxasyncclient-instance

And: https://www.python-httpx.org/advanced/timeouts/

Create an `httpx.AsyncClient()` with the timeout settings you desire and pass it as the `client` to `GraphClientFactory.create_with_default_middleware()`.
